src/neuroconv/datainterfaces/behavior/video/externalvideointerface.py

```diff
@@ -0,0 +1,345 @@
+        parent_container: {'acquisition', 'processing/behavior'}
+            The container where the ImageSeries is added, default is nwbfile.acquisition.
+            When 'processing/behavior' is chosen, the ImageSeries is added to nwbfile.processing['behavior'].
+        module_description: str, optional
+            If parent_container is 'processing/behavior', and the module does not exist,
+            it will be created with this description. The default description is the same as used by the
+            conversion_tools.get_module function.
+        """
+        metadata = metadata or dict()
+
+        file_paths = self.source_data["file_paths"]
+
+        # Be sure to copy metadata at this step to avoid mutating in-place
+        videos_metadata = deepcopy(metadata).get("Behavior", dict()).get("Video", None)
+        if videos_metadata is None:
+            videos_metadata = deepcopy(self.get_metadata()["Behavior"]["Video"])
+        image_series_kwargs = metadata["Behavior"]["Video"][self.video_name]
+        image_series_kwargs["name"] = self.video_name
+
+        timing_type = self.get_timing_type()
+        if self._number_of_files > 1 and starting_frames is None:
+            raise TypeError("Multiple paths were specified for the ImageSeries, but no starting_frames were specified!")
+        elif starting_frames is not None and len(starting_frames) != self._number_of_files:
+            raise ValueError(
+                f"Multiple paths ({self._number_of_files}) were specified for the ImageSeries, "
+                f"but the length of starting_frames ({len(starting_frames)}) did not match the number of paths!"
+            )
+        elif starting_frames is not None:
+            image_series_kwargs.update(starting_frame=starting_frames)
+
+        image_series_kwargs.update(format="external", external_file=file_paths)
+
+        if timing_type == "starting_time and rate":
+            starting_time = self._segment_starting_times[0] if self._segment_starting_times is not None else 0.0
+            with VideoCaptureContext(file_path=str(file_paths[0])) as video:
+                rate = video.get_video_fps()
+            image_series_kwargs.update(starting_time=starting_time, rate=rate)
+        elif timing_type == "timestamps":
+            image_series_kwargs.update(timestamps=np.concatenate(self._timestamps))
+        else:
+            raise ValueError(f"Unrecognized timing_type: {timing_type}")
+
+        # Attach image series
+        image_series = ImageSeries(**image_series_kwargs)
+        if parent_container == "acquisition":
+            nwbfile.add_acquisition(image_series)
+        elif parent_container == "processing/behavior":
+            get_module(nwbfile=nwbfile, name="behavior", description=module_description).add(image_series)
+        else:
+            raise ValueError(
+                f"parent_container must be either 'acquisition' or 'processing/behavior', not {parent_container}."
```

Comment:
As per your comment in other PRs validation logic should be at the start of the function, what do you think?

Reply:
Yep, thanks for the catch.